magenta-lib/src/main/scala/magenta/tasks/ChangeSetTasks.scala

```diff
@@ -0,0 +1,119 @@
+package magenta.tasks
+
+import com.amazonaws.services.cloudformation.model.{Change, DeleteChangeSetRequest, DescribeChangeSetRequest, ExecuteChangeSetRequest}
+import com.amazonaws.services.s3.AmazonS3
+import magenta.artifact.S3Path
+import magenta.tasks.UpdateCloudFormationTask._
+import magenta.{DeployReporter, KeyRing, Region}
+
+import scala.collection.JavaConverters._
+
+class CreateChangeSetTask(
+   region: Region,
+   templatePath: S3Path,
+   stackLookup: CloudFormationStackLookup,
+   val unresolvedParameters: CloudFormationParameters
+)(implicit val keyRing: KeyRing, artifactClient: AmazonS3) extends Task {
+
+  override def execute(reporter: DeployReporter, stopFlag: => Boolean) = if (!stopFlag) {
+    val cfnClient = CloudFormation.makeCfnClient(keyRing, region)
+    val s3Client = S3.makeS3client(keyRing, region)
+    val stsClient = STS.makeSTSclient(keyRing, region)
+    val accountNumber = STS.getAccountNumber(stsClient)
+
+    val templateString = templatePath.fetchContentAsString.right.getOrElse(
+      reporter.fail(s"Unable to locate cloudformation template s3://${templatePath.bucket}/${templatePath.key}")
+    )
+
+    val (stackName, changeSetType) = stackLookup.lookup(reporter, cfnClient)
+
+    val template = processTemplate(stackName, templateString, s3Client, stsClient, region, reporter)
+    val parameters = unresolvedParameters.resolve(template, accountNumber, changeSetType, reporter, cfnClient)
+
+    reporter.info("Creating Cloudformation change set")
+    reporter.info(s"Stack name: $stackName")
+    reporter.info(s"Change set name: ${unresolvedParameters.changeSetName}")
```

Comment:
Does changeSetName really need to be in this `CloudFormationParameters` object. It's already available in the `CloudFormationStackLookup` object in this scope. I don't believe it is used outside of here.